azurelinuxagent/common/protocol/wire.py

```diff
@@ -1304,6 +1330,70 @@ def parse(self, xml_text):
         return self
 
 
+class RemoteAccess(object):
+    """
+    Object containing information about user accounts
+    """
+    #
+    # <RemoteAccess>
+    #   <Version/>
+    #   <Incarnation/>
+    #    <Users>
+    #       <User>
+    #         <Name/>
+    #         <Password/>
+    #         <Expiration/>
+    #       </User>
+    #     </Users>
+    #   </RemoteAccess>
+    #
+
+    def __init__(self, xml_text):
+        logger.verbose("Load RemoteAccess.xml")
+        self.version = None
+        self.incarnation = None
+        self.user_list = RemoteAccessUsersList()
+
+        self.xml_text = None
+        self.parse(xml_text)
+
+    def parse(self, xml_text):
+        """
+        Parse xml document containing user account information
+        """
+        if xml_text is None or len(xml_text) == 0:
+            return None
+        self.xml_text = xml_text
+        xml_doc = parse_doc(xml_text)
+        self.incarnation = findtext(xml_doc, "Incarnation")
+        self.version = findtext(xml_doc, "Version")
+        user_collection = find(xml_doc, "Users")
+        users = findall(user_collection, "User")
+
+        for user in users:
+            remote_access_user = self.parse_user(user)
+            self.user_list.users.append(remote_access_user)
+        return self
+
+    def parse_user(self, user):
```

Comment:
unless this is a sticking point, I'd rather not make this change now since the code is stable.  I'll look at changing this when adding delete.

---
In reply to: [194519470](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r194519470) [](ancestors = 194519470)